### ISSUE TITLE
Save: Continuously update to reflect most recent changes in master

### ISSUE THREAD

**Issue by ellynliu** [uchicago-cs/chiventure#219]
Whenever game state pushes to master (including the hotfix for iterator functions), if necessary, update the .proto file AND update save.h and save.c files to reflect changes. This could include changes in structs, new functions, function names, parameters, etc.

**Comment by jarvislam1999** [comment]
I updated game.proto, save.h and save.c in the following ways:
- Updated struct in game.proto to reflect master, write save functions for the new structs
- Commented out the iterator functions and use the new functions for save
- Update Makefile to import action-management code

Next steps:
- Update the tests to be able to test save.c
- Update load.c to reflect master

**Comment by jarvislam1999** [comment]
I have successfully updated the save module along with the tests such that all the tests for save pass. This means that the save code is ready for master, since the compilation is error free, plus no segmentation fault. However, the test team for save should make the tests more detailed during the rest of the sprint to make sure that the save module works seemlessly.

Summary of the work for save:
- Updated tests to test for lengths of hashtable
- Make sure every proto structs were properly initialized
- Some control flow errors. 

I may close this issue soon since the code is up to date with master, but will leave it open in case master changes again.

**Comment by jarvislam1999** [comment]
I have also successfully updated the load module along with the tests such that all the tests for load pass. Again, the test team for load should make the tests more detailed during the rest of the sprint. I posted the work for load here since the original issue for load was closed and that I was not assigned to it.

Summary of the work for load:

- Updated tests to test for lengths of hashtable, and also initialize game struct so we could load on it
- Make sure every proto structs were properly initialized
- Get rid of infinite loop and NULL pointer errors.
